### ISSUE TITLE
Failed to normalize ... maybe try to call `try_normalize_erasing_regions` instead

### ISSUE THREAD

**Issue by Mark-Simulacrum** [rust-lang/rust#114425]
* https://crater-reports.s3.amazonaws.com/beta-1.72-2/beta-2023-07-23/reg/abi_stable-0.11.2/log.txt

```
[INFO] [stdout] error: internal compiler error: compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:195:90: Failed to normalize unsafe extern "C" fn(sabi_types::rref::RRef<'_, marker_type::ErasedObject>) -> std_types::result::RResult<<I as erased_types::traits::GetSerializeProxyType<'_>>::ProxyType, std_types::std_error::RBoxError_>, maybe try to call `try_normalize_erasing_regions` instead
```

**Comment by Nilstrieb** [comment]
not having tested it myself, this is probably #113900 which was fixed by #113921
It was reported for this crate. The PR seems safe to backport if deemed important enough.

**Comment by cuviper** [comment]
#113921 was backported in #114748.
